### ISSUE TITLE
Activity endpoint - CREATE

### ISSUE THREAD

**Issue by paulgibbs** [buddypress/BP-REST#25]
Needs building.

Consider allowing regular users to only create activity_update's. Or not.
Either way, we'll need comprehensive permission checks to make sure people can't, for example, create activity items inside private/hidden groups.

**Comment by robruiz** [comment]
I have been trying to figure out how to implement this with little progress. 

`register_rest_route( $this->namespace, '/' . $this->rest_base, array(
			array(
				// Trying to create Activity Feeds via API
				'methods'  => WP_REST_Server::CREATABLE,
				// Register Callback function for Creating Activities
				'callback' => array( $this, 'create_activity' ),
				// Args for endpoint -> we need a user_id, some body to work with, a component, and a type
				'permission_callback' => array( $this, 'get_items_permissions_check' ),
				'args'     => array($this, 'get_new_activity_args'),
			),
			'schema' => array( $this, 'get_public_item_schema' ),
		) );`

**Comment by modemlooper** [comment]
Hi @robruiz,

I'm not sure where your level of coding knowledge is at, this repo is not a place to learn how to create API's. However, we welcome all pull requests. If you're not familiar with the WordPress REST API I'd suggest starting there. Learn about extending the core REST API. Then you will need knowledge of adding BuddyPress activity items. Check these two links to get started. 

[](http://v2.wp-api.org/extending/internal-classes/)
[](https://codex.buddypress.org/developer/function-examples/bp_activity_add/)

**Comment by renatonascalves** [comment]
Here are the scenarios where a user can create:

- A user needs to be logged_in.
- Admins can create everywhere.
- Moderators as well.
- A user is a member of the group (group component needs to be active).

The last one fixes the problem of private/hidden groups. Users can only post in groups they are a member of.

**Comment by robruiz** [comment]
I ended up writing something that works. However, we ended up moving away from Buddypress to manage these notifications and created our own system. Buddypress just seemed to specific to the platform and we are only using bits and pieces of the platform. It wasn't too difficult, but it did not include permission checks or anything, just simple oAuth1.0 implementation. If anyone is interested in seeing the code, I would be more than happy to share, but I doubt you all are interested.
